sympy/solvers/solveset.py

```diff
@@ -534,64 +534,104 @@ def _solve_as_rational(f, symbol, domain):
 
 def _solve_trig(f, symbol, domain):
     """Function to call other helpers to solve trigonometric equations """
-    sol1 = sol = None
+    sol = None
     try:
-        sol1 = _solve_trig1(f, symbol, domain)
-    except NotImplementedError:
-        pass
-    if sol1 is None or isinstance(sol1, ConditionSet):
+        sol = _solve_trig1(f, symbol, domain)
+    except (ValueError, NotImplementedError):
         try:
             sol = _solve_trig2(f, symbol, domain)
         except ValueError:
-            sol = sol1
-        if isinstance(sol1, ConditionSet) and isinstance(sol, ConditionSet):
-            if sol1.count_ops() < sol.count_ops():
-                sol = sol1
-    else:
-        sol = sol1
-    if sol is None:
-        raise NotImplementedError(filldedent('''
-            Solution to this kind of trigonometric equations
-            is yet to be implemented'''))
+            raise NotImplementedError(filldedent('''
+                Solution to this kind of trigonometric equations
+                is yet to be implemented'''))
     return sol
 
 
 def _solve_trig1(f, symbol, domain):
-    """Primary helper to solve trigonometric and hyperbolic equations"""
+    """Primary helper to solve trigonometric and hyperbolic equations
+
+    Either returns solution set as ConditionSet (evaluated or unevaluated)
+    or raises NotImplementedError.
+
+    """
+    # Prepare change of variable
+    x = Dummy('x')
     if _is_function_class_equation(HyperbolicFunction, f, symbol):
-        cov = exp(symbol)
+        cov = exp(x)
         inverter = invert_real if domain.is_subset(S.Reals) else invert_complex
     else:
-        cov = exp(I*symbol)
+        cov = exp(I*x)
         inverter = invert_complex
+
     f = trigsimp(f)
     f_original = f
+    trig_functions = f.atoms(TrigonometricFunction, HyperbolicFunction)
+    trig_arguments = [e.args[0] for e in trig_functions]
+    # trigsimp may have reduced the equation to an expression
+    # that is independent of 'symbol' (e.g. cos**2+sin**2)
+    if not any(a.has(symbol) for a in trig_arguments):
+        return solveset(f_original, symbol, domain)
+
+    denominators = []
+    numerators = []
+    for ar in trig_arguments:
+        try:
+            poly_ar = Poly(ar, symbol)
+        except ValueError:
+            raise ValueError("give up, we can't solve if this is not a polynomial in x")
```

Comment:
When does Poly raise ValueError?

If these ValueErrors are intended to be caught above then it would be better to make a special exception class for it like:
```python
class _SolveTrig1Error(Exception):
    """Raised when solvetrig1 heuristics do not apply"""
    pass
```
If the exception class is local to these helper functions then there is no need for it to be part of any hierarchy or public API.

Reply:
> When does Poly raise ValueError?

Hmm, good question. I copied it over from `_solve_trig2` and thought it was to harden the code a bit if for some reason a non-polynomial ends up as an argument to a trig function. (This shouldn't happen if called via `solveset`, but maybe it was intended to make it more self-contained and guard against user error.)
But now I notice that it actually raises `PolynomialError` for obviously non-polynomial arguments and not `ValueError`. I'll grep through Poly code and probably change it to PolynomialError.

Reply:
I introduced `_SolveTrig1Error` as suggested.